File: app/src/main/kotlin/es/voghdev/playbattlegrounds/common/ui/ColoredSnackbar.java
package es.voghdev.playbattlegrounds.common.ui;

import android.graphics.Color;
import android.graphics.Typeface;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.TextView;

public class ColoredSnackbar {

    private static final int red = 0xfff44336;
    private static final int green = 0xff4caf50;
    private static final int blue = 0xff2195f3;
    private static final int orange = 0xffffc107;

    private static View getSnackBarLayout(Snackbar snackbar) {
        if (snackbar != null) {
            return snackbar.getView();
        }
        return null;
    }

    private static Snackbar colorSnackBar(Snackbar snackbar, int colorId) {
        return colorSnackBar(snackbar, colorId, false);
    }

    private static Snackbar colorSnackBar(Snackbar snackbar, int colorId, boolean bold) {
        View snackBarView = getSnackBarLayout(snackbar);
        if (snackBarView != null) {
            TextView snackBarTextView = snackBarView.findViewById(android.support.design.R.id.snackbar_text);
            snackBarView.setBackgroundColor(colorId);
            snackBarTextView.setTypeface(null, bold ? Typeface.BOLD : Typeface.NORMAL);
            snackBarTextView.setTextColor(Color.WHITE);
        }

        return snackbar;
    }

    public static Snackbar info(Snackbar snackbar) {
        return colorSnackBar(snackbar, blue);
    }

    public static Snackbar warning(Snackbar snackbar) {
        return colorSnackBar(snackbar, orange);
    }

    public static Snackbar warningBold(Snackbar snackbar) {
        return colorSnackBar(snackbar, orange, true);
    }

    public static Snackbar alert(Snackbar snackbar) {
        return colorSnackBar(snackbar, red);
    }

    public static Snackbar alertBold(Snackbar snackbar) {
        return colorSnackBar(snackbar, red, true);
    }

    public static Snackbar confirm(Snackbar snackbar) {
        return colorSnackBar(snackbar, green);
    }
}

File: app/src/main/kotlin/es/voghdev/playbattlegrounds/common/ui/ColoredSnackbar.kt
package es.voghdev.playbattlegrounds.common.ui

import android.graphics.Color
import android.graphics.Typeface
import android.support.design.widget.Snackbar
import android.view.View
import android.widget.TextView

object ColoredSnackbar {

    private const val red = 0xff_f4_43_36.toInt()
    private const val green = 0xff_4c_af_50.toInt()
    private const val blue = 0xff_21_95_f3.toInt()
    private const val orange = 0xff_ff_c1_07.toInt()

    private fun getSnackBarLayout(snackbar: Snackbar?): View? = snackbar?.view

    private fun colorSnackBar(snackbar: Snackbar, colorId: Int, bold: Boolean = false): Snackbar {
        val snackBarView = getSnackBarLayout(snackbar)
        if (snackBarView != null) {
            val snackBarTextView = snackBarView.findViewById<TextView>(android.support.design.R.id.snackbar_text)
            snackBarView.setBackgroundColor(colorId)
            snackBarTextView.setTypeface(null, if (bold) Typeface.BOLD else Typeface.NORMAL)
            snackBarTextView.setTextColor(Color.WHITE)
        }

        return snackbar
    }

    fun info(snackbar: Snackbar): Snackbar = colorSnackBar(snackbar, blue)

    fun warning(snackbar: Snackbar): Snackbar = colorSnackBar(snackbar, orange)

    fun warningBold(snackbar: Snackbar): Snackbar = colorSnackBar(snackbar, orange, true)

    fun alert(snackbar: Snackbar): Snackbar = colorSnackBar(snackbar, red)

    fun alertBold(snackbar: Snackbar): Snackbar = colorSnackBar(snackbar, red, true)

    fun confirm(snackbar: Snackbar): Snackbar = colorSnackBar(snackbar, green)
}
